datasync/chaindatafetcher/kafka/consumer.go

```diff
@@ -306,37 +307,99 @@ func (c *Consumer) updateOffset(buffer [][]*Segment, lastMsg *sarama.ConsumerMes
 	return nil
 }
 
+// renewExpireMsg updates expiration timer and message if the expiring message is handled.
+func (c *Consumer) renewExpireMsg(buffer [][]*Segment, timer *time.Timer, oldestMsg *sarama.ConsumerMessage) *sarama.ConsumerMessage {
+	if c.config.ExpirationTime > time.Duration(0) {
+		if len(buffer) == 0 {
+			timer.Stop()
+			return nil
+		} else if oldestMsg != buffer[0][0].orig {
```

Comment:
Is `buffer` is affected when timer.Reset is executed? If not, what about changing the code to return only `buffer[0][0].orig`?
```
if oldestMsg != buffer[0][0].orig {
    timer.Reset(c.config.ExpirationTime)
}
return buffer[0][0].orig
```

Reply:
@yoomee1313 The code is updated, could you take a look once more?

Reply:
@jeongkyun-oh How about returning buffer[0][0].orig instead of returning oldMsg? It would be helpful to understand the logic easily. As long as I understood, buffer[0][0].orig is always the oldest msg, right?

Reply:
@yoomee1313 Oh, they may not be the same. So, if they are the same, it returns `buffer[0][0].orig` without resetting the timer. Otherwise, it returns `oldestMsg` with restting the timer.